File: problems/Two Pointers/ContainerWithMostWater.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        n = len(height)
        l = 0
        r = n-1
        cap = 0

        while l < r:
            width = r-l
            if(height[l] <= height[r]):
                currentCap = height[l]*width
                l += 1
            else:
                currentCap = height[r]*width
                r -= 1
            if(currentCap > cap):
                cap = currentCap
        return cap

File: problems/Java/ContainerWithMostWater.java
class Solution {
	public int maxArea(int[] height) {
		int n = height.length;
		int l = 0;
		int r = n - 1;
		int capacity = 0;
		int currentCap = 0;

		while (l < r) {
			int width = r - l;
			if (height[l] > height[r]) {
				currentCap = height[r] * width;
				r--;
			} else {
				currentCap = height[l] * width;
				l++;
			}

			capacity = Math.max(currentCap, capacity);

		}
		return capacity;
	}
}